lib/fizzy/stack.hpp

```diff
@@ -13,47 +13,39 @@
 namespace fizzy
 {
 template <typename T>
-class Stack : public std::vector<T>
+class Stack
 {
-public:
-    using difference_type = typename std::vector<T>::difference_type;
-
-    using std::vector<T>::vector;
-
-    using std::vector<T>::back;
-    using std::vector<T>::emplace_back;
-    using std::vector<T>::pop_back;
-    using std::vector<T>::resize;
-    using std::vector<T>::size;
+    std::vector<T> m_container;
 
-    // Also used: size(), resize(), clear(), empty(), end()
-
-    void push(T val) { emplace_back(val); }
+public:
+    void push(T val) { m_container.emplace_back(val); }
 
     template <typename... Args>
     void emplace(Args&&... args)
     {
-        std::vector<T>::emplace_back(std::forward<Args>(args)...);
+        m_container.emplace_back(std::forward<Args>(args)...);
     }
 
-    T pop()
+    T pop() noexcept
```

Comment:
Is this marked noexcept because we manually ensure the caller will call it in a bad state?

Reply:
Assert is not relevant. This is `noexcept` because `.back()` and `.pop_back()` are `noexcept` too.

Reply:
I checked it when I commented on empty/size, but to me back/pop_back did not seem to be noexcept:
- http://cplusplus.com/reference/vector/vector/back/
- http://cplusplus.com/reference/vector/vector/pop_back/

Reply:
Ah okay noexcept since C++20:
- https://en.cppreference.com/w/cpp/container/vector/back
- https://en.cppreference.com/w/cpp/container/vector/pop_back

Reply:
Were these behaving as noexcept pre-C++20 just not marked so?

Reply:
Officially they are not `noexcept` (even in C++20), but in practice they are.

Reply:
This looks good to me.